added error message when catalog has wrong api key

ywkim312 opened the pull request:
## Description
This should give the proper error message in the log when the published documents are not showing on the browser without logging in. @sandeep-ps Since I am out, you can modify and push necessary changes if needed please

**Fixes #923**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

minump reviewed (changes requested):
Not able to see published contributions when not logged in. The error remains the same:
"2022-06-02T17:13:44.130Z INFO    [Thread-35 ] : root - Contribution SEARCH {"search": "Contribution search performed with : None", "ep_building_block": "contributions_building_block", "ep_name": "Contribution", "ep_method": "SEARCH"}
"

ywkim312 commented:
> Not able to see published contributions when not logged in. The error remains the same: "2022-06-02T17:13:44.130Z INFO [Thread-35 ] : root - Contribution SEARCH {"search": "Contribution search performed with : None", "ep_building_block": "contributions_building_block", "ep_name": "Contribution", "ep_method": "SEARCH"} "

This PR is not the PR for showing the published contribution without logging in. If you want me to do include it, I will create an issue and include it in here.

> Not able to see published contributions when not logged in. The error remains the same: "2022-06-02T17:13:44.130Z INFO [Thread-35 ] : root - Contribution SEARCH {"search": "Contribution search performed with : None", "ep_building_block": "contributions_building_block", "ep_name": "Contribution", "ep_method": "SEARCH"} "

ywkim312 commented:
> Not able to see published contributions when not logged in. The error remains the same: "2022-06-02T17:13:44.130Z INFO [Thread-35 ] : root - Contribution SEARCH {"search": "Contribution search performed with : None", "ep_building_block": "contributions_building_block", "ep_name": "Contribution", "ep_method": "SEARCH"} "

I have checked with my local and it shows the published contributions without logging in. Are you sure it doesn't show the published ones without logging in?

ywkim312 commented:
> Not able to see published contributions when not logged in. The error remains the same: "2022-06-02T17:13:44.130Z INFO [Thread-35 ] : root - Contribution SEARCH {"search": "Contribution search performed with : None", "ep_building_block": "contributions_building_block", "ep_name": "Contribution", "ep_method": "SEARCH"} "

@minump Can you post your published contributions Json in here to see if it is correct?

ywkim312 commented on contributions/catalog/catalog_rest_service.py:
```diff
@@ -114,6 +115,14 @@ def index():
+                "ERROR": "There is a problem getting contributions list. Maybe API Key is not correct."
+            }
+            msg_json = jsonutil.create_log_json("Contribution", "SEARCH", msg)
+            logging.info("Contribution SEARCH " + json.dumps(msg))
```
This doesn't make any difference because msg will be overridden as the same as msg_json but I will change it to msg_json for clearer understanding of the code.

ywkim312 commented:
Added the message when there is no published contribution saying that there is no published contribution. Also, added the redirection to error pages when the api key is wrong from just showing main page with empty contributions.

minump commented:
> > Not able to see published contributions when not logged in. The error remains the same: "2022-06-02T17:13:44.130Z INFO [Thread-35 ] : root - Contribution SEARCH {"search": "Contribution search performed with : None", "ep_building_block": "contributions_building_block", "ep_name": "Contribution", "ep_method": "SEARCH"} "
> 
> @minump I have checked with my local and it shows the published contributions without logging in. Can we double check to see if it doesn't show the published ones without logging in?

Able to see published contributions when not logged in. Must have been a mistake from my end before. 
Also able to see "There is no published contributions" in home.html when using wrong APIKey.

sandeep-ps commented on contributions/catalog/webapps/templates/contribute/home.html:
```diff
@@ -249,6 +249,12 @@ <h4 class="talent_name">{{ talent|filter_nested_dict(["name"]) }}</h4>
+
+            {% if (tal_json|length == 0) and (cap_json|length == 0)  %}
+                <div>
+                    <h4 class="talent_name">There is no published contributions.</h4>
```
```suggestion
                    <h4 class="talent_name">There are no published Contributions.</h4>
```

sandeep-ps commented:
I will move the CHANGELOG entry to the unreleased section. The rest looks good.

ywkim312 commented:
> I will move the CHANGELOG entry to the unreleased section. The rest looks good.

thanks

sandeep-ps commented:
@minump, it looks like you had requested some changes in this PR. If they look good, could you please approve this PR? Thanks.

sandeep-ps commented:
Thank you! Approved and merged.
